recipes/logr/v0.7/conanfile.py

```diff
@@ -0,0 +1,133 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import get, copy,rm
+from conan.tools.layout import basic_layout
+from conan.tools.scm import Version
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc
+import os
+
+required_conan_version = ">=1.50.0"
+
+class LogrConan(ConanFile):
+    name = "logr"
+    description = (
+        "Logger frontend substitution for spdlog, glog, etc "
+        "for server/desktop applications"
+    )
+    license = "BSD-3-Clause"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/ngrodzitski/logr"
+    topics = ("logger", "development", "util", "utils")
+    settings = "os", "arch", "compiler", "build_type"
+    package_type = "header-library"
+    options = {
+        "with_spdlog_backend": [True, False],
+        "with_glog_backend": [True, False],
+        "with_log4cplus_backend": [True, False],
+        "with_boostlog_backend": [True, False],
+    }
+    default_options = {
+        "with_spdlog_backend": True,
+        "with_glog_backend": False,
+        "with_log4cplus_backend": False,
+        "with_boostlog_backend": False,
+    }
+
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("fmt/10.2.1")
+
+        if self.options.with_spdlog_backend:
+            self.requires("spdlog/1.12.0", transitive_headers=True, transitive_libs=True)
+
+        if self.options.with_glog_backend:
+            self.requires("glog/0.6.0", transitive_headers=True, transitive_libs=True)
+
+        if self.options.with_log4cplus_backend:
+            self.requires("log4cplus/2.1.0", transitive_headers=True, transitive_libs=True)
+
+        if self.options.with_boostlog_backend:
+            self.requires("boost/1.83.0", transitive_headers=True, transitive_libs=True)
```

Comment:
```suggestion
        if self.options.with_spdlog_backend:
            self.requires("spdlog/1.12.0")

        if self.options.with_glog_backend:
            self.requires("glog/0.6.0")

        if self.options.with_log4cplus_backend:
            self.requires("log4cplus/2.1.0")

        if self.options.with_boostlog_backend:
            self.requires("boost/1.83.0")
```

As per https://docs.conan.io/2/reference/conanfile/methods/requirements.html#package-type-trait-inferring, a `package_type = "header-library"` recipe will have those traits inferred by default :)